### ISSUE TITLE
Failed to serialize yolov7-w6/d6/e6/e6e to plan file for kernel weights has count 0 but 65280 was expected

### ISSUE THREAD

**Issue by daileyet** [wang-xinyu/tensorrtx#1196]
## Env

- GPU, NVIDIA RTX A6000
- OS, Ubuntu 20.04
- Cuda version，11.7
- TensorRT version,  22.07

## About this repo

- which model? yolov7 w6/e6/d6/e6e

## Your problem

- what is your command? 
get train mode best.pt base on yolov7-w6.pt
```shell
python /workspace/yolov7/train_aux.py --batch-size 16 --epochs 5 --project /workspace/yolov7/runs/train --data /workspace/cv_datasets/coco128/coco128.yaml --img 1280 1280 --cfg /workspace/yolov7/cfg/training/yolov7-w6.yaml --weights /workspace/yolov7/yolov7-w6.pt --hyp /workspace/yolov7/data/hyp.scratch.p6.yaml
```
```shell
python gen_wts.py -w /workspace/yolov7/runs/train/exp/weights/best.pt -o /workspace/yolov7/runs/train/exp/weights/best.wts
/workspace/tensorrtx/yolov7/build/yolov7 -s yolov7/runs/train/exp/weights/best.wts yolov7/runs/train/exp/weights/best.engine w6
```
- what's your output?
```
[100%] Linking CXX executable yolov7
[100%] Built target yolov7
Loading weights: runs/train/exp/weights/best.wts
[01/07/2023-10:15:41] [W] [TRT] The implicit batch dimension mode has been deprecated. Please create the network with NetworkDefinitionCreationFlag::kEXPLICIT_BATCH flag whenever possible.
Building engine, please wait for a while...
[01/07/2023-10:15:41] [E] [TRT] 3: cv105.0:kernel weights has count 0 but 65280 was expected
[01/07/2023-10:15:41] [E] [TRT] 4: cv105.0: count of 0 weights in kernel, but kernel dimensions (1,1) with 256 input channels, 255 output channels and 1 groups were specified. Expected Weights count is 256 * 1*1 * 255 / 1 = 65280
[01/07/2023-10:15:41] [E] [TRT] 4: [convolutionNode.cpp::computeOutputExtents::43] Error Code 4: Internal Error (cv105.0: number of kernel weights does not match tensor dimensions)
[01/07/2023-10:15:41] [E] [TRT] 4: [network.cpp::validate::2881] Error Code 4: Internal Error (Could not compute dimensions for (Unnamed Layer* 0) [Slice]_output, because the network is not valid.)
[01/07/2023-10:15:41] [E] [TRT] 2: [builder.cpp::buildSerializedNetwork::636] Error Code 2: Internal Error (Assertion engine != nullptr failed. )
Build engine successfully!
yolov7: /workspace/tensorrtx/yolov7/main.cpp:38: void serialize_engine(unsigned int, std::string&, std::string&, std::string&): Assertion `serialized_engine != nullptr' failed.

```
- what output do you expect?
should convert wts to engine successfully same as yolov7 model v7/x.

Does anybody know how to resolve this issue?
Thanks a lot.

**Comment by wang-xinyu** [comment]
@daileyet Can you try the pretrained weight first?

**Comment by xyt2008** [comment]
遇见同样的问题，原因是用最新的yolov7训练的，不是用v0.1版本训练的，最新的转的时候报这个错。

**Comment by daileyet** [comment]
> @daileyet Can you try the pretrained weight first?

Yeah, I try convert pretrained w6 weights, it is OK. But myself train model base on w6 is error.

**Comment by daileyet** [comment]
> 遇见同样的问题，原因是用最新的yolov7训练的，不是用v0.1版本训练的，最新的转的时候报这个错。

好的， 我再试试，但p5models转没有问题，p6models就转不成功。

**Comment by daileyet** [comment]
> 



> 遇见同样的问题，原因是用最新的yolov7训练的，不是用v0.1版本训练的，最新的转的时候报这个错。

@xyt2008 请教一下，你说的原因是 预训练模型yolov7-w6.pt是最新的，不是v1.0的？
还是用的是最新的代码训练的模型转换有问题，用v0.1的代码训练的转换是没有问题的？

多谢。

**Comment by daileyet** [comment]
Solved as below steps

1. checkout yolov7 v0.1 code
2. train use pre-trained model with  v0.1

p5model
```shell
python /workspace/yolov7-0.1/train.py --batch-size 16 --epochs 5 --project /workspace/yolov7-0.1/runs/train --data /workspace/coco128/coco128.yaml --img-size 640 --cfg /workspace/yolov7-0.1/cfg/deploy/yolov7.yaml --weights /workspace/premodels/yolov7-0.1/yolov7.pt --hyp /workspace/yolov7-0.1/data/hyp.scratch.p5.yaml
```
p6model
```shell
python /workspace/yolov7-0.1/train.py --batch-size 16 --epochs 5 --project /workspace/yolov7-0.1/runs/train --data /workspace/coco128/coco128.yaml --img-size 1280 --cfg /workspace/yolov7-0.1/cfg/deploy/yolov7-w6.yaml --weights /workspace/premodels/yolov7-0.1/yolov7-w6.pt --hyp /workspace/yolov7-0.1/data/hyp.scratch.p6.yaml
```

3. convert from wts to engine

**Comment by xyt2008** [comment]
> 

用最新的转不了，会报错。用v0.1是可以转的，但是我用v0.1训练精度特别差，不知道什么原因。

**Comment by xyt2008** [comment]
> 

用最新的转不了，会报错。用v0.1是可以转的，但是我用v0.1训练精度特别差，不知道什么原因。
